app/modules/template/service.server.ts

```diff
@@ -0,0 +1,240 @@
+        },
+        data: {
+          status: AssetStatus.AVAILABLE,
+        },
+      });
+
+      // Send notifications
+      await createNote({
+        content: `The PDF template **${updatedTemplate.name}** now requires a signature. **${custody.custodian.name}** needs to sign the **${updatedTemplate.name}** template before receiving custody.`,
+        type: "UPDATE",
+        userId,
+        assetId: custody.assetId,
+      });
+    }
+  }
+
+  return updateTemplate;
+}
+
+export async function updateTemplatePDF({
+  request,
+  pdfName,
+  pdfSize,
+  templateId,
+  organizationId,
+}: {
+  request: Request;
+  templateId: string;
+  pdfName: string;
+  pdfSize: number;
+  organizationId: User["id"];
+}) {
+  const res = await db.template.findFirst({
+    where: { id: templateId, organizationId },
+    select: { name: true, pdfUrl: true },
+  });
+
+  if (!res) return null;
+
+  const newFileName: string = `${organizationId}/${templateId}`;
+  const fileData = await parseFileFormData({
+    request,
+    bucketName: "templates",
+    newFileName,
+    updateExisting: res.pdfUrl !== null && res.pdfUrl !== undefined,
+  });
+
+  const pdf = fileData.get("pdf") as string;
+
+  if (!pdf) return null;
+
+  const publicUrl = getPublicFileURL({
+    bucketName: "templates",
+    filename: newFileName,
+  });
+
+  const data = {
+    pdfUrl: publicUrl + ".pdf",
+    pdfSize,
+    pdfName,
+  };
+
+  return db.template.update({
+    where: { id: templateId, organizationId },
+    data,
+  });
+}
+
+export async function makeInactive({
+  id,
+  organizationId,
+}: Pick<Template, "id"> & { organizationId: Organization["id"] }) {
+  return db.template.update({
+    where: { id, organizationId },
+    data: {
+      isActive: false,
+      isDefault: false,
+    },
+  });
+}
+
+export async function makeActive({
+  id,
+  organizationId,
+}: Pick<Template, "id"> & { organizationId: Organization["id"] }) {
+  return db.template.update({
+    where: { id, organizationId },
+    data: {
+      isActive: true,
+    },
+  });
+}
+
+export async function makeDefault({
+  id,
+  type,
+  organizationId,
+}: {
+  id: Template["id"];
+  type: Template["type"];
+  organizationId: Organization["id"];
+}) {
+  // Make all the templates of the same type of the user non-default
+  await db.template.updateMany({
+    where: { type, organizationId },
+    data: { isDefault: false },
+  });
+
+  // Make the selected template default
+  return db.template.update({
+    where: { id, organizationId },
+    data: { isDefault: true },
+  });
+}
+
+export async function getTemplateById({ id }: Pick<Template, "id">) {
+  return db.template.findFirst({
+    where: { id },
+  });
+}
+
+export async function getTemplates({
+  organizationId,
+  page = 1,
+  perPage = 8,
+}: {
+  organizationId: Organization["id"];
+  page?: number;
+  perPage?: number;
+}) {
+  const where = {
+    organizationId,
+  };
+
+  const [templates, totalTemplates] = await Promise.all([
```

Comment:
Just curious, why not use transactions from prisma?

Reply:
I was not familiar with prisma tx back then, will do it now

Reply:
Resolved

Reply:
OKey I am sorry, but we have to reverse that. In one of the latest things we did, we basically did some performance testing and Promise.all is 30-70% faster than using prisma's transaction. The change should be pretty easy tho